werkzeug/_reloader.py

```diff
@@ -197,10 +204,23 @@ def run(self):
                     try:
                         watches[path] = observer.schedule(
                             self.event_handler, path, recursive=True)
-                    except OSError:
-                        # "Path is not a directory". We could filter out
-                        # those paths beforehand, but that would cause
-                        # additional stat calls.
+                    except OSError as e:
+                        # Extract message from exception
+                        message = str(e)
+
+                        # TODO: These are hardcoded, should find better way to differentiate errors from watchdog
+                        if message == "inotify watch limit reached":
+                            # Warn user to increase limit for inotify watches
+                            _log('error', 'inotify limit reached, please increase your limit')
+                        elif message == "Path is not a directory":
+                            # "Path is not a directory". We could filter out
+                            # those paths beforehand, but that would cause
+                            # additional stat calls.
+                            _log('error', '%s is not a directory' % path)
+                        else:
+                            # Otherwise, log the exception
+                            _log('error', str(e))
```

Comment:
* We can use `message` here
* Why special-case "inotify watch limit reached" if the new error message is almost identical?